packages/middleware/__tests__/unit/apiClientFactory/apiClientFactory.spec.ts

```diff
@@ -76,11 +78,11 @@ describe("apiClientFactory", () => {
     };
 
     const { createApiClient } = apiClientFactory<any, any>(params as any);
-    const extensions = (createApiClient as any)._predefinedExtensions;
+    const extensions = ((await createApiClient) as any)._predefinedExtensions;
 
-    createApiClient.bind({ middleware: { req: null, res: null, extensions } })(
-      {}
-    );
+    await createApiClient.bind({
```

Comment:
Since you're calling this function immediately anyway, perhaps you could replace `.bind() ` with [.call()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call)?